client/lib/browser-storage/index.ts

```diff
@@ -146,6 +186,34 @@ export async function getStoredItem< T >( key: string ): Promise< T | undefined
 	return await idbGet( key );
 }
 
+/**
+ * Get all stored items.
+ *
+ * @param pattern The pattern to match on returned item keys.
+ * @returns A promise with the stored key/value pairs as an object. Empty if none.
+ */
+export async function getAllStoredItems( pattern?: RegExp ): Promise< StoredItems > {
+	if ( shouldBypass ) {
+		return await bypassGetAll( pattern );
+	}
+
+	const idbSupported = await supportsIDB();
+	if ( ! idbSupported ) {
+		const results: StoredItems = {};
+		for ( let i = 0; i < window.localStorage.length; i++ ) {
+			const key = window.localStorage.key( i );
+			if ( ! key || ( pattern && ! pattern?.test( key ) ) ) {
```

Comment:
Another unneeded optional chaining? You just checked if `pattern` is truthy and yet use `pattern?.test`.

Reply:
Right, totally unnecessary.